src/OneDimensionalDeviate.cpp

```diff
@@ -78,56 +79,77 @@ namespace galsim {
+            xdbg<<"fatleft = "<<fatLeft<<"  "<<std::abs(df1)<<" <? "<<std::abs(df2)<<std::endl;
+            // Loop invariants:
+            xassert(x1 < x2);
+            xassert(x2 < x3);
+            xassert(df1 == f2 - f1);
+            xassert(df2 == f3 - f2);
+            xassert(df1 * df2 < 0.);
+            xassert(fatLeft == (std::abs(df1) < std::abs(df2)));
+
             if (fatLeft) {
+                xdbg<<"fat left\n";
                 // Split left-hand interval
                 double xTrial = x1 + GOLDEN*(x2-x1);
                 double fTrial = function(xTrial);
                 double dfTrial = f2 - fTrial;
-                if (df1 * dfTrial <= 0.) {
+                xdbg<<"Trial = "<<xTrial<<","<<fTrial<<","<<dfTrial<<std::endl;
+                if (dfTrial * df2 < 0.) {
+                    xdbg<<"trial/2/3\n";
+                    // Extremum is in trial / 2 / 3
+                    x1 = xTrial;
+                    f1 = fTrial;
+                    df1 = dfTrial;
+                } else {
+                    xdbg<<"1/trial/2\n";
                     // Now bracketed in 1 / trial / 2
                     x3 = x2;
                     f3 = f2;
                     x2 = xTrial;
                     f2 = fTrial;
                     df1 = f2 - f1;
                     df2 = dfTrial;
-                    fatLeft = true;
-                } else {
-                    // Extremum is in trial / 2 / 3
-                    x1 = xTrial;
-                    f1 = fTrial;
-                    df1 = dfTrial;
-                    fatLeft = false;
                 }
             } else {
+                xdbg<<"fat right\n";
                 // Split right-hand interval (2 / trial / 3)
-                double xTrial = x2 - GOLDEN*(x3-x2);
+                double xTrial = x3 - GOLDEN*(x3-x2);
                 double fTrial = function(xTrial);
                 double dfTrial = fTrial - f2;
-                if (dfTrial * df2 <= 0.) {
+                xdbg<<"Trial = "<<xTrial<<","<<fTrial<<","<<dfTrial<<std::endl;
+                if (dfTrial * df1 < 0.) {
+                    xdbg<<"1/2/trial\n";
+                    // Extremum is in 1 / 2 / trial
+                    x3 = xTrial;
+                    f3 = fTrial;
+                    df2 = dfTrial;
+                } else {
+                    xdbg<<"2/trial/3\n";
                     // Now bracketed in 2 / trial / 3
                     x1 = x2;
                     f1 = f2;
                     x2 = xTrial;
                     f2 = fTrial;
                     df1 = dfTrial;
                     df2 = f3 - f2;
-                    fatLeft = false;
-                } else {
-                    // Extremum is in 1 / 2 / trial
-                    x3 = xTrial;
-                    f3 = fTrial;
-                    df2 = dfTrial;
-                    fatLeft = true;
                 }
             }
+            fatLeft = std::abs(df1) < std::abs(df2);
```

Comment:
This didn't used to be preserved in the old version.  But I think not preserving this could only cause a slight inefficiency, not cause the code to fail.